app/org/sagebionetworks/bridge/models/schedules/CriteriaScheduleStrategy.java

```diff
@@ -0,0 +1,198 @@
+package org.sagebionetworks.bridge.models.schedules;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+
+import org.springframework.validation.Errors;
+
+import org.sagebionetworks.bridge.models.Criteria;
+import org.sagebionetworks.bridge.models.CriteriaUtils;
+import org.sagebionetworks.bridge.util.BridgeCollectors;
+import org.sagebionetworks.bridge.validators.ScheduleValidator;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Sets;
+
+public final class CriteriaScheduleStrategy implements ScheduleStrategy {
+    
+    private final List<ScheduleCriteria> scheduleCriteria = Lists.newArrayList();
+
+    public void addCriteria(ScheduleCriteria criteria) {
+        this.scheduleCriteria.add(criteria);
+    }
+    
+    public List<ScheduleCriteria> getScheduleCriteria() {
+        return scheduleCriteria;
+    }
+    
+    public void setScheduleCriteria(List<ScheduleCriteria> criteria) {
+        this.scheduleCriteria.clear();
+        if (criteria != null) {
+            this.scheduleCriteria.addAll(criteria);    
+        }
+    }    
+    
+    /**
+     * Iterate through the list of schedules and return the first schedule that matches 
+     * the criteria submitted for this user and this request (so order matters). Can 
+     * return null, and this is accounted for now elsewehere in the code.
+     * @param plan
+     * @param context
+     * @return schedule that matches users's criteria, or null if no schedules match the 
+     *      criteria.
+     */
+    @Override
+    public Schedule getScheduleForUser(SchedulePlan plan, ScheduleContext context) {
+        for (ScheduleCriteria criteria : scheduleCriteria) {
+            if (CriteriaUtils.matchCriteria(context, criteria)) {
+                return criteria.getSchedule();
+            }
+        }
+        return null;        
+    }
+
+    @Override
+    public void validate(Set<String> dataGroups, Set<String> taskIdentifiers, Errors errors) {
+        for (int i=0; i < scheduleCriteria.size(); i++) {
+            ScheduleCriteria criteria = scheduleCriteria.get(i);
+            errors.pushNestedPath("scheduleCriteria["+i+"]");
+            if (criteria.getSchedule() == null){
+                errors.rejectValue("schedule", "is required");
```

Comment:
Yes, I had to remove it in order to get the case where it is null to show up in the validation errors. Otherwise you get a null pointer exception/a 500 error. So now this is verified as part of validation.